### ISSUE TITLE
Trouble loading osmFISH.ipynb

### ISSUE THREAD

**Issue by ambrosejcarr** [spacetx/starfish#520]
@simone-codeluppi noticed a problem with loading the osmFISH data. 

Here's a replication of the bug, executing on (current) master@732f6de: 

I tried restarting my computer in case other programs were contributing to the open files. 

In the notebook: 
```python
experiment = Experiment.from_json("https://dmf0bdeheu4zf.cloudfront.net/20180828/osmFISH/experiment.json")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/projects/spacetx/slicedimage/slicedimage/io.py in resolve_url(name_or_url, baseurl, allow_caching)

~/projects/spacetx/slicedimage/slicedimage/io.py in _resolve_absolute_url(absolute_url, allow_caching)

~/projects/spacetx/slicedimage/slicedimage/io.py in infer_backend(baseurl, allow_caching)

ValueError: Unable to infer backend for url , please verify that baseurl points to a valid directory or web address

During handling of the above exception, another exception occurred:


During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
/usr/local/lib/python3.6/site-packages/IPython/core/interactiveshell.py in showtraceback(self, exc_tuple, filename, tb_offset, exception_only, running_compiled_code)
   1862                         # in the engines. This should return a list of strings.
-> 1863                         stb = value._render_traceback_()
   1864                     except Exception:

AttributeError: 'OperationalError' object has no attribute '_render_traceback_'

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
/usr/local/lib/python3.6/site-packages/IPython/core/interactiveshell.py in run_code(self, code_obj, result)
   2978             if result is not None:
   2979                 result.error_in_exec = sys.exc_info()[1]
-> 2980             self.showtraceback(running_compiled_code=True)
   2981         else:
   2982             outflag = False

/usr/local/lib/python3.6/site-packages/IPython/core/interactiveshell.py in showtraceback(self, exc_tuple, filename, tb_offset, exception_only, running_compiled_code)
   1864                     except Exception:
   1865                         stb = self.InteractiveTB.structured_traceback(etype,
-> 1866                                             value, tb, tb_offset=tb_offset)
   1867 
   1868                     self._showtraceback(etype, value, stb)

/usr/local/lib/python3.6/site-packages/IPython/core/ultratb.py in structured_traceback(self, etype, value, tb, tb_offset, number_of_lines_of_context)

/usr/local/lib/python3.6/site-packages/IPython/core/ultratb.py in structured_traceback(self, etype, value, tb, tb_offset, number_of_lines_of_context)

/usr/local/lib/python3.6/site-packages/IPython/core/ultratb.py in structured_traceback(self, etype, evalue, etb, tb_offset, number_of_lines_of_context)

TypeError: must be str, not list
```

While that is executing, you get the following jupyter (backend) error continuously until it dies:
```python
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 916, in _bootstrap_inner
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 864, in run
  File "/usr/local/lib/python3.6/site-packages/ipykernel/iostream.py", line 78, in _thread_main
  File "/usr/local/lib/python3.6/site-packages/tornado/platform/asyncio.py", line 112, in start
    self.asyncio_loop.run_forever()
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/base_events.py", line 422, in run_forever
    self._run_once()
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/base_events.py", line 1432, in _run_once
    handle._run()
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/events.py", line 156, in _run
    self._loop.call_exception_handler(context)
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/base_events.py", line 1297, in call_exception_handler
    exc_info=True)
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/logging/__init__.py", line 1335, in error
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/logging/__init__.py", line 1442, in _log
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/logging/__init__.py", line 1452, in handle
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/logging/__init__.py", line 1514, in callHandlers
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/logging/__init__.py", line 863, in handle
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/logging/__init__.py", line 998, in emit
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/logging/__init__.py", line 915, in handleError
  File "/usr/local/lib/python3.6/site-packages/ipykernel/iostream.py", line 376, in write
  File "/usr/local/lib/python3.6/site-packages/ipykernel/iostream.py", line 203, in schedule
  File "/usr/local/lib/python3.6/site-packages/ipykernel/iostream.py", line 101, in _event_pipe
  File "/usr/local/lib/python3.6/site-packages/zmq/sugar/context.py", line 146, in socket
  File "/usr/local/lib/python3.6/site-packages/zmq/sugar/socket.py", line 58, in __init__
  File "zmq/backend/cython/socket.pyx", line 330, in zmq.backend.cython.socket.Socket.__init__
zmq.error.ZMQError: Too many open files

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/ipykernel/iostream.py", line 97, in _event_pipe
AttributeError: '_thread._local' object has no attribute 'event_pipe'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 884, in _bootstrap
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 926, in _bootstrap_inner
  File "/usr/local/lib/python3.6/site-packages/ipykernel/iostream.py", line 376, in write
  File "/usr/local/lib/python3.6/site-packages/ipykernel/iostream.py", line 203, in schedule
  File "/usr/local/lib/python3.6/site-packages/ipykernel/iostream.py", line 101, in _event_pipe
  File "/usr/local/lib/python3.6/site-packages/zmq/sugar/context.py", line 146, in socket
  File "/usr/local/lib/python3.6/site-packages/zmq/sugar/socket.py", line 58, in __init__
  File "zmq/backend/cython/socket.pyx", line 330, in zmq.backend.cython.socket.Socket.__init__
zmq.error.ZMQError: Too many open files
IOStream.flush timed out
```

**Comment by ambrosejcarr** [comment]
The .py script runs successfully for me: 
```
python3 notebooks/py/osmFISH.py 
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 78/78 [01:37<00:00,  1.25s/it]
interactive(children=(IntSlider(value=0, description='plane_index', max=38), Output()), _dom_classes=('widget-interact',))

```

**Comment by ambrosejcarr** [comment]
My python environment: 
```
$ pip3 freeze
alabaster==0.7.10
anndata==0.5.5
apipkg==1.4
appdirs==1.4.3
appnope==0.1.0
argcomplete==1.9.4
argh==0.26.2
asn1crypto==0.24.0
atomicwrites==1.1.5
attrs==18.1.0
awscli==1.14.58
Babel==2.5.3
backcall==0.1.0
bcrypt==3.1.4
beautifulsoup4==4.6.0
bhtsne==0.1.9
biopython==1.71
bleach==2.1.3
boto3==1.6.11
botocore==1.9.11
cachetools==2.0.1
cairocffi==0.8.0
CairoSVG==2.1.3
certifi==2018.1.18
cffi==1.11.5
cftime==1.0.1
chardet==3.0.4
click==6.7
clickclick==1.2.2
cloudpickle==0.5.3
colorama==0.3.7
connexion==1.1.15
coverage==4.5.1
crcmod==1.7
Crimson==0.4.0
cromwell-manager==0.0.1
-e git+https://github.com/broadinstitute/cromwell-tools.git@fd14e3c72c606d943dcc12400f3f6ce74d7dc666#egg=cromwell_tools
cryptography==2.1.4
cssselect==1.0.1
cssselect2==0.2.1
cvxopt==1.1.9
cycler==0.10.0
Cython==0.28
dask==0.18.1
decorator==4.3.0
defusedxml==0.5.0
dill==0.2.7.1
diskcache==3.0.6
docopt==0.6.2
docutils==0.14
doubletdetection==2.0.0
ecdsa==0.13
entrypoints==0.2.3
enum34==1.1.6
et-xmlfile==1.0.1
execnet==1.5.0
fastcluster==1.1.24
feather-format==0.3.1
-e git+https://github.com/ambrosejcarr/fin_annotate.git@482af2fede07eb0beeda7f1ee65aae11e8016e7d#egg=fin_annotate
flake8==3.5.0
flake8-import-order==0.18
Flask==0.12.2
future==0.16.0
gapic-google-cloud-datastore-v1==0.15.3
gapic-google-cloud-error-reporting-v1beta1==0.15.3
gapic-google-cloud-logging-v2==0.91.3
gapic-google-cloud-pubsub-v1==0.15.4
gapic-google-cloud-spanner-admin-database-v1==0.15.3
gapic-google-cloud-spanner-admin-instance-v1==0.15.3
gapic-google-cloud-spanner-v1==0.15.3
gffutils==0.9
google-auth==1.1.1
google-auth-oauthlib==0.1.1
google-cloud==0.27.0
google-cloud-bigquery==0.26.0
google-cloud-bigtable==0.26.0
google-cloud-core==0.26.0
google-cloud-datastore==1.2.0
google-cloud-dns==0.26.0
google-cloud-error-reporting==0.26.0
google-cloud-language==0.27.0
google-cloud-logging==1.2.0
google-cloud-monitoring==0.26.0
google-cloud-pubsub==0.27.0
google-cloud-resource-manager==0.26.0
google-cloud-runtimeconfig==0.26.0
google-cloud-spanner==0.26.0
google-cloud-speech==0.28.0
google-cloud-storage==1.3.2
google-cloud-translate==1.1.0
google-cloud-videointelligence==0.25.0
google-cloud-vision==0.26.0
google-gax==0.15.15
google-resumable-media==0.2.3
googleapis-common-protos==1.5.2
gprof2dot==2017.9.19
grpc-google-iam-v1==0.11.4
grpcio==1.4.0
gseapy==0.7.9
h5py==2.7.1
hca==1.1.3
html5lib==1.0.1
httplib2==0.10.3
idna==2.6
imagesize==1.0.0
inflection==0.3.1
intervaltree==2.1.0
ipykernel==4.8.2
ipython==6.4.0
ipython-genutils==0.2.0
ipywidgets==7.1.2
itsdangerous==0.24
jdcal==1.3
jedi==0.12.0
Jinja2==2.10
jmespath==0.9.3
joblib==0.11
jsonpath-rw==1.4.0
jsonpointer==1.10
jsonschema==2.6.0
jupyter==1.0.0
jupyter-client==5.2.3
jupyter-console==5.2.0
jupyter-contrib-core==0.3.1
jupyter-contrib-nbextensions==0.2.7
jupyter-core==4.4.0
jupyter-highlight-selected-word==0.0.11
jupyter-latex-envs==1.3.8.4
jupyter-nbextensions-configurator==0.2.4
jupyterlab==0.32.1
jupyterlab-launcher==0.10.5
kiwisolver==1.0.1
llvmlite==0.24.0
loompy==2.0.8
lxml==3.7.3
MarkupSafe==1.0
matplotlib==2.2.3
mccabe==0.6.1
memory-profiler==0.52.0
mistune==0.8.3
mock==2.0.0
monotonic==1.3
more-itertools==4.2.0
multiprocessing-on-dill==3.5.0a4
mypy==0.620
natsort==5.2.0
nbconvert==5.3.1
nbconvert-pyencoded==5.3.2.dev1
nbencdec==0.0.6
nbformat==4.4.0
networkx==2.1
nose==1.3.7
nose2==0.7.4
notebook==5.4.1
numba==0.39.0
numexpr==2.6.4
numpy==1.14.5
oauth2client==4.1.2
oauthlib==2.0.4
olefile==0.44
opencv-python==3.4.0.12
openpyxl==2.4.9
packaging==17.1
pandas==0.23.4
pandocfilters==1.4.2
paramiko==2.4.1
parso==0.2.1
patsy==0.5.0
pbr==4.1.0
pdfrw==0.4
pexpect==4.6.0
PhenoGraph==1.5.2
pickleshare==0.7.4
Pillow==5.0.0
PIMS==0.4.1
pkginfo==1.4.1
pluggy==0.7.1
ply==3.8
pockets==0.6.2
profilehooks==1.10.0
prompt-toolkit==1.0.15
proto-google-cloud-datastore-v1==0.90.4
proto-google-cloud-error-reporting-v1beta1==0.15.3
proto-google-cloud-logging-v2==0.91.3
proto-google-cloud-pubsub-v1==0.15.4
proto-google-cloud-spanner-admin-database-v1==0.15.3
proto-google-cloud-spanner-admin-instance-v1==0.15.3
proto-google-cloud-spanner-v1==0.15.3
protobuf==3.4.0
psutil==5.4.3
ptyprocess==0.5.2
py==1.5.3
pyasn1==0.3.4
pyasn1-modules==0.1.5
pycodestyle==2.3.1
pycparser==2.18
pycrypto==2.6.1
Pydap==3.2.2
pyfaidx==0.5.3.1
pyflakes==1.6.0
Pygments==2.2.0
pygtrie==2.2
Pyment==0.3.2
PyNaCl==1.2.1
pyparsing==2.2.0
Pyphen==0.9.4
pysam==0.14.1
pytest==3.7.2
pytest-cov==2.5.1
pytest-flake8==1.0.1
pytest-forked==0.2
pytest-mock==1.10.0
pytest-profiling==1.3.0
pytest-xdist==1.22.5
python-dateutil==2.7.2
pytz==2018.3
PyWavelets==0.5.2
PyYAML==3.12
pyzmq==17.0.0
qtconsole==4.3.1
regex==2018.2.21
regional==1.1.2
requests==2.19.1
requests-mock==1.4.0
requests-oauthlib==0.8.0
requests-toolbelt==0.8.0
rsa==3.4.2
-e git+https://github.com/ambrosejcarr/rut.git@e969744542fc24689d482fd286e5d56644298093#egg=rut
s3transfer==0.1.13
scanpy==0.4.4
scikit-image==0.14.0
scikit-learn==0.19.2
scipy==1.1.0
sctools==0.1.10.dev4+gb3a3c98.d20180513
seaborn==0.8.1
semantic-version==2.6.0
Send2Trash==1.5.0
seqc==0.2.0
setuptools-scm==2.0.0
showit==1.1.4
simplegeneric==0.8.1
simplejson==3.13.2
six==1.11.0
sklearn==0.0
-e git+https://github.com/spacetx/slicedimage.git@bf2f19b0185b00a12d8102d18d69a19e9f21b77b#egg=slicedimage
slicerator==0.9.8
snowballstemmer==1.2.1
sortedcontainers==1.5.9
Sphinx==1.7.2
sphinx-rtd-theme==0.3.0
sphinxcontrib-napoleon==0.6.1
sphinxcontrib-websupport==1.0.1
-e git+https://github.com/spacetx/starfish.git@732f6de351610bfe61691969fe27769933e913dc#egg=starfish
statsmodels==0.8.0
suffix-trees==0.2.4.4
swagger-spec-validator==2.1.0
tables==3.4.2
tenacity==4.10.0
terminado==0.8.1
testpath==0.3.1
-e git+https://github.com/ambrosejcarr/timesheet.git@beaf8043c2e38703df1ca63c2f27deb8e70a9193#egg=timesheet
tinycss==0.4
tinycss2==0.6.1
tinydb==3.8.0
toolz==0.9.0
tornado==5.0.1
tqdm==4.25.0
trackpy==0.4.1
traitlets==4.3.2
tweak==0.5.1
twine==1.9.1
typed-ast==1.1.0
typing==3.6.4
umap==0.1.1
umap-learn==0.3.2
urllib3==1.22
-e git+https://github.com/spacetx/sptx-format@7968957f02463f13d63237038cdd6e5e489b8c79#egg=validate_sptx
validators==0.12.2
wcwidth==0.1.7
WeasyPrint==0.42.2
webencodings==0.5.1
WebOb==1.8.2
Werkzeug==0.12.2
widgetsnbextension==3.1.4
wikipedia==1.4.0
xarray==0.10.8
xlrd==1.1.0
```

**Comment by ambrosejcarr** [comment]
Closed by 96b6b8d
